src/accelerate/utils.py

```diff
@@ -163,16 +163,7 @@ def recursively_apply(func, data, *args, test_type=is_torch_tensor, error_on_oth
                 for o in data
             ),
         )
-    elif isinstance(data, UserDict):
-        return type(data)(
-            {
-                k: recursively_apply(
-                    func, v, *args, test_type=test_type, error_on_other_type=error_on_other_type, **kwargs
-                )
-                for k, v in data.items()
-            }
-        )
-    elif isinstance(data, dict):
+    elif isinstance(data, Mapping):
         return type(data)(
             **{
```

Comment:
You would need to remove the `**` for this to work with UserDict I believe.
```suggestion
            {
```

Reply:
`UserDict` also works with the `**` syntax. You can test this with the following code or by looking at the source [here](https://github.com/python/cpython/blob/2c56c97f015a7ea81719615ddcf3c745fba5b4f3/Lib/collections/__init__.py#L964-L984) (warning you that the code is a bit messy in Python 3.6-3.8 compared to Python 3.9):
```python
from collections import UserDict
assert UserDict(**{"a": 1, "b": 2}) == UserDict({"a": 1, "b": 2})
```

Reply:
But not the `BatchEncoding` subclass we are doing all of this for :-)

You can try and see that:
```
from transformers import BatchEncoding
assert BatchEncoding(**{"a": 1, "b": 2}) == BatchEncoding({"a": 1, "b": 2})
```
fails

Reply:
Oh, I see. Thanks.